### PR TITLE
MK/MIN-319

### PR DESCRIPTION
**Pull request by l-polina**
**Description:**

Implemented connection to the polkadot.js extension. Received and displayed data: Lend Balance, Borrow balance, Total Lent, Total Borrow and Collateral Ratio.

**Screenshots:**

image
image
image

### CODE REVIEW

File: src/components/Markets/BorrowMarkets/index.tsx [hunk 1]
```diff
@@ -10,9 +14,7 @@ import DOT from '../../../assets/images/image 9.png';
 import './styles.scss';
 
 export default function BorrowMarkets(props: BorrowMarketsProps) {
-	const { rates, handleBorrow } = props;
-
-	const data = [[3342.76, 959.24, 376.29, 134.25]];
+	const { rates, handleBorrow, userBorrowPerAsset, currentPrice } = props;
 
 	const assetsList = ['BTC', 'ETH', 'DOT', 'KSM'];
```
Comment (gavrilin-remote) [inline src/components/Markets/BorrowMarkets/index.tsx:19]: move constant out of Functional Component

File: src/components/Markets/BorrowMarkets/index.tsx [hunk 2]
```diff
@@ -32,8 +34,18 @@ export default function BorrowMarkets(props: BorrowMarketsProps) {
...
+		const totalBorrowPerAsset =
+			userBorrowPerAsset &&
+			formatBalanceData(userBorrowPerAsset[asset].value.amount);
+
+		const currentPricePerAsset =
+			currentPrice &&
+			parseFloat((currentPrice[asset].toString() / 10 ** 18).toString());
+
+		const totalBorrowInUSD = currentPricePerAsset * totalBorrowPerAsset;
+
```
Comment (gavrilin-remote) [inline src/components/Markets/BorrowMarkets/index.tsx:46]: memoize values totalBorrowPerAsset, currentPricePerAsset and totalBorrowInUSD with useMemo hook.

File: src/components/Markets/LendMarkets/index.tsx [hunk 3]
```diff
@@ -10,9 +14,8 @@ import DOT from '../../../assets/images/image 9.png';
 import './styles.scss';
 
 export default function LendMarkets(props: LendMarketsProps) {
-	const { rates, handleLend } = props;
-
-	const data = [[9991.56, 3926.46, 941.26, 337.46]];
+	const { rates, handleLend, userUnderlyingBalancePerAsset, currentPrice } =
+		props;
 
 	const assetsList = ['BTC', 'ETH', 'DOT', 'KSM'];
```
Comment (gavrilin-remote) [inline src/components/Markets/LendMarkets/index.tsx:20]: move constant out of Functional Component

File: src/components/Markets/LendMarkets/index.tsx [hunk 4]
```diff
@@ -32,8 +35,18 @@ export default function LendMarkets(props: LendMarketsProps) {
...
+		const totalLentPerAsset =
+			userUnderlyingBalancePerAsset &&
+			formatBalanceData(userUnderlyingBalancePerAsset[asset].value.amount);
+
+		const currentPricePerAsset =
+			currentPrice &&
+			parseFloat((currentPrice[asset].toString() / 10 ** 18).toString());
+
+		const totalLentInUSD = currentPricePerAsset * totalLentPerAsset;
+
```
Comment (gavrilin-remote) [inline src/components/Markets/LendMarkets/index.tsx:47]: memoize values totalLentPerAsset, currentPricePerAsset and totalLentInUSD with useMemo hook.

File: src/components/Markets/MarketUserBalance/index.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+	const borrowBalance = userBalanceInfo
+		? formatBalanceData(userBalanceInfo.total_borrowed)
+		: '0';
+
+	const totalCollateral = accountCollateral
+		? formatBalanceData(accountCollateral.value.amount)
+		: '0';
+
+	const collateralRatio =
```
Comment (gavrilin-remote) [inline src/components/Markets/MarketUserBalance/index.tsx:51]: memoize this function with useMemo

File: src/components/Widgets/CollateralRatioWidget/index.tsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import './style.scss';
+
+export default function CollateralRatioWidget(collateralRatio: any) {
+	const coef = 135 / 50;
```
Comment (gavrilin-remote) [inline src/components/Widgets/CollateralRatioWidget/index.tsx:5]: move constants out of Functional Component

**Review by gavrilin-remote — CHANGES_REQUESTED** [review]

File: src/components/Markets/BorrowMarkets/index.tsx (hunk 1 above)
Comment (l-polina) [reply src/components/Markets/BorrowMarkets/index.tsx]: > move constant out of Functional Component
The asset list is sourced from BE.

File: src/components/Markets/LendMarkets/index.tsx (hunk 3 above)
Comment (l-polina) [reply src/components/Markets/LendMarkets/index.tsx]: > move constant out of Functional Component

The asset list is sourced from BE.

File: src/components/Markets/LendMarkets/index.tsx (hunk 4 above)
Comment (l-polina) [reply src/components/Markets/LendMarkets/index.tsx]: > memoize values totalLentPerAsset, currentPricePerAsset and totalLentInUSD with useMemo hook
memoized values ​​totalLentPerAsset, currentPricePerAsset and totalLentInUSD with useMemo hook

File: src/components/Markets/BorrowMarkets/index.tsx (hunk 2 above)
Comment (l-polina) [reply src/components/Markets/BorrowMarkets/index.tsx]: > memoize values totalBorrowPerAsset, currentPricePerAsset and totalBorrowInUSD with useMemo hook.

memoized values totalBorrowPerAsset, currentPricePerAsset and totalBorrowInUSD with useMemo hook

File: src/components/Markets/MarketUserBalance/index.tsx (hunk 5 above)
Comment (l-polina) [reply src/components/Markets/MarketUserBalance/index.tsx]: > memoize this function with useMemo

memoized this function with useMemo

File: src/components/Widgets/CollateralRatioWidget/index.tsx (hunk 6 above)
Comment (l-polina) [reply src/components/Widgets/CollateralRatioWidget/index.tsx]: > move constants out of Functional Component

moved constant

**Review by gavrilin-remote — APPROVED** [review]

**Pull request merged**